### PR TITLE
Merge authorization header with options

### LINKED ISSUE #643: Pass additional headers into Octokit::Client::Authorizations#scope
Currently the [scopes method](https://github.com/octokit/octokit.rb/blob/master/lib/octokit/client/authorizations.rb#L118) doesn't allow us to pass additional headers for the request.

My use case is that I want to set the

```ruby
'Cache-Control' => 'no-cache, no-store'
```

header so that the response isn't cached.

Is this a change that would be welcome?

/cc @johndbritton

### PR DESCRIPTION
**Pull request by tarebyte**
Closes #643, this allows the user to pass additional options to the `scopes` method.

Feedback welcome

/cc @johndbritton @pengwynn

### CODE REVIEW

File: lib/octokit/client/authorizations.rb [hunk 1]
```diff
@@ -115,10 +115,17 @@ def delete_authorization(number, options = {})
       # @param token [String] GitHub OAuth token
```
Comment (johndbritton) [inline lib/octokit/client/authorizations.rb]: > @param options [Hash] ...

File: lib/octokit/client/authorizations.rb [hunk 2]
```diff
@@ -113,12 +113,20 @@ def delete_authorization(number, options = {})
...
-      def scopes(token = @access_token)
+      def scopes(token = @access_token, options = {})
         raise ArgumentError.new("Access token required") if token.nil?
+        authorization_header = { "Authorization" => "token #{token}" }
+
+        if options[:headers]
+          options[:headers].merge(authorization_header)
+        else
+          options[:headers] = authorization_header
+        end
```
Comment (pengwynn) [inline lib/octokit/client/authorizations.rb]: Just a style preference, I think I'd tackle this like:

```ruby
auth = { "Authorization" => "token #{token}" }
headers = (options.delete(:headers) || {}).merge(auth)

agent.call(:get, "user", :headers => headers).

**Comment by tarebyte** [comment]
I think this is ready to go unless anyone else has anymore feedback :smile:

**Comment by pengwynn** [comment]
:cake:

**Comment by tarebyte** [comment]
Thanks @pengwynn :heart_eyes:

**Pull request merged**